US369215: Disable weaker security algorithms

michael-bryson opened the pull request:
@dermot-hardy The patch part is not working and I cannot figure it out, have tried many different combinations so hoping fresh eyes might see where I am going wrong.

May need to switch to updating the crypto-policies: https://en.opensuse.org/SDB:Crypto-policies

Commit:
US369215: Disable insecure TLS algorithms

Commit:
Update

buildmachine-sou-jenkins2 commented:
CI Build Link:
http://sou-jenkins2.swinfra.net:8080/job/CAFapi/job/CAFapi~opensuse-java17-images~US369215~CI

dermot-hardy committed:
Corrected patch file syntax

This will just make the same change that is being made for Java 11 but whether or not that is the correct change needs to be studied.

Commit:
Added other algorithms

Commit:
Removed unnecessary algorithms from patch

michael-bryson commented:
@dermot-hardy @rob-duewer I have made the changes to the algorithm list to match what was in the email Rob sent last week. However I am confused because when I use the tomcat image from [here](https://github.com/CAFapi/opensuse-tomcat-jre11-image/pull/28) in the tag-service build [here](https://github.houston.softwaregrp.net/Verity/tag-service/pull/310), it still reports the same allowed algorithms as before any patching was done.

[java_17_no_sec.txt](https://github.com/CAFapi/opensuse-java17-images/files/8927771/java_17_no_sec.txt)

dermot-hardy commented:
I took a look and I can't see any obvious issues.  @kusumaghoshdastidar I've re-assigned this ticket to you for investigation.

kusumaghoshdastidar committed:
Add more algorithms to disable list

kusumaghoshdastidar committed:
Update disableWeakTlsAlgorithms.patch

kusumaghoshdastidar committed:
Set crypto policies to FUTURE level: https://en.opensuse.org/SDB:Crypto-policies

kusumaghoshdastidar committed:
Remove unnecessary file

kusumaghoshdastidar committed:
Remove script once policies are updated

kusumaghoshdastidar committed:
Move system wide crypto policy change to base image

kusumaghoshdastidar committed:
Clean up

kusumaghoshdastidar committed:
Update base image version

kusumaghoshdastidar committed:
Clean repos
